ra-optimisation/search-tree-rao/src/test/java/com/powsybl/openrao/searchtreerao/marmot/MarmotTest.java

```diff
@@ -159,4 +164,63 @@ void testWithPreventiveTopologicalAction() throws IOException {
+        Crac crac9 = Crac.read("/crac/crac-202503251830.json", MarmotTest.class.getResourceAsStream("/crac/crac-202503251830.json"), network9);
+        Crac crac10 = Crac.read("/crac/crac-202503251930.json", MarmotTest.class.getResourceAsStream("/crac/crac-202503251930.json"), network10);
+
+        RaoParameters raoParameters = JsonRaoParameters.read(MarmotTest.class.getResourceAsStream("/parameters/RaoParameters_minCost_megawatt_dc.json"));
+
+        OffsetDateTime timestamp1 = OffsetDateTime.of(2025, 3, 25, 10, 30, 0, 0, ZoneOffset.UTC);
+        OffsetDateTime timestamp2 = OffsetDateTime.of(2025, 3, 25, 11, 30, 0, 0, ZoneOffset.UTC);
+        OffsetDateTime timestamp3 = OffsetDateTime.of(2025, 3, 25, 12, 30, 0, 0, ZoneOffset.UTC);
+        OffsetDateTime timestamp4 = OffsetDateTime.of(2025, 3, 25, 13, 30, 0, 0, ZoneOffset.UTC);
+        OffsetDateTime timestamp5 = OffsetDateTime.of(2025, 3, 25, 14, 30, 0, 0, ZoneOffset.UTC);
+        OffsetDateTime timestamp6 = OffsetDateTime.of(2025, 3, 25, 15, 30, 0, 0, ZoneOffset.UTC);
+        OffsetDateTime timestamp7 = OffsetDateTime.of(2025, 3, 25, 16, 30, 0, 0, ZoneOffset.UTC);
+        OffsetDateTime timestamp8 = OffsetDateTime.of(2025, 3, 25, 17, 30, 0, 0, ZoneOffset.UTC);
+        OffsetDateTime timestamp9 = OffsetDateTime.of(2025, 3, 25, 18, 30, 0, 0, ZoneOffset.UTC);
+        OffsetDateTime timestamp10 = OffsetDateTime.of(2025, 3, 25, 19, 30, 0, 0, ZoneOffset.UTC);
+
+        Map<OffsetDateTime, RaoInput> inputPerTimestamp = new HashMap<>();
+        inputPerTimestamp.put(timestamp1, RaoInput.build(network1, crac1).build());
+        inputPerTimestamp.put(timestamp2, RaoInput.build(network2, crac2).build());
+        inputPerTimestamp.put(timestamp3, RaoInput.build(network3, crac3).build());
+        inputPerTimestamp.put(timestamp4, RaoInput.build(network4, crac4).build());
+        inputPerTimestamp.put(timestamp5, RaoInput.build(network5, crac5).build());
+        inputPerTimestamp.put(timestamp6, RaoInput.build(network6, crac6).build());
+        inputPerTimestamp.put(timestamp7, RaoInput.build(network7, crac7).build());
+        inputPerTimestamp.put(timestamp8, RaoInput.build(network8, crac8).build());
+        inputPerTimestamp.put(timestamp9, RaoInput.build(network9, crac9).build());
+        inputPerTimestamp.put(timestamp10, RaoInput.build(network10, crac10).build());
+
+        InterTemporalRaoInput input = new InterTemporalRaoInput(
+            new TemporalDataImpl<>(inputPerTimestamp),
+            Set.of(GeneratorConstraints.create().withGeneratorId("FFR1AA1 _generator").withLeadTime(0.0).withLagTime(0.0).withPMin(0.0).withPMax(5000.0).withUpwardPowerGradient(500.0).withDownwardPowerGradient(-500.0).build())
+        );
+
+        InterTemporalRaoResultImpl interTemporalRaoResult = (InterTemporalRaoResultImpl) new Marmot().run(input, raoParameters).join();
+
+        assertEquals(625070.0, interTemporalRaoResult.getGlobalFunctionalCost(InstantKind.PREVENTIVE));
```

Comment:
Maybe you could add an assert on the hourly setpoint of the redispatching action too.

Reply:
Also duplicate the test without the GeneratorConstraints to see the difference